staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -443,19 +444,20 @@ func (r *crdHandler) getOrCreateServingInfoFor(crd *apiextensions.CustomResource
 		typer := newUnstructuredObjectTyper(parameterScheme)
 		creator := unstructuredCreator{}
 
-		validator, _, err := apiservervalidation.NewSchemaValidator(crd.Spec.Validation)
+		validationSchema := getCRDSchemaForVersionOrDie(crd, v.Name)
+		validator, _, err := apiservervalidation.NewSchemaValidator(validationSchema)
 		if err != nil {
 			return nil, err
 		}
 
 		var statusSpec *apiextensions.CustomResourceSubresourceStatus
 		var statusValidator *validate.SchemaValidator
-		if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) && crd.Spec.Subresources != nil && crd.Spec.Subresources.Status != nil {
-			statusSpec = crd.Spec.Subresources.Status
-
+		subresources := getCRDSubresourcesForVersionOrDie(crd, v.Name)
```

Comment:
not sure why OrDie variants are being used here... why wouldn't we return an error?

Reply:
Are you against all DoXOrDie methods or only here? What is the difference?

Reply:
using them for programming errors is still a little questionable when there's an easy way to propagate an error, but is more reasonable.

this is using them to fail on data errors, which I think is a bad idea.